RepositoryCompiler/CodeModel/CodeParsers/CSharp/CaDETClassMetricCalculator.cs

```diff
@@ -123,5 +124,43 @@ private int GetNumberOfMethodsDeclared(CaDETClass parsedClass)
         {
             return parsedClass.Members.Count(method => method.Type.Equals(CaDETMemberType.Method));
         }
+
+        /// <summary>
+        /// CC - Class cohesion
+        /// DOI: 10.14257/ijmue.2014.9.6.35
+        /// </summary>
+        private double? GetCCClassCohesion(CaDETClass parsedClass)
+        {
+            IEnumerable<CaDETMember> memberMethods = parsedClass.Members.Where(m => m.Type != CaDETMemberType.Property);
+            IEnumerable<CaDETMember> memberProperties = parsedClass.Members.Where(m => m.Type == CaDETMemberType.Property);
+            double fieldsCount = parsedClass.Fields.Count + memberProperties.Count();
+
+            if (fieldsCount == 0 || memberMethods.Count() == 0)
+            {
+                return 0;
+            }
+
+            double cv = CalculateCohesionForEveryVariable(parsedClass, memberMethods, memberProperties);
+
+            return Math.Round(cv / fieldsCount, 2);
+        }
+
+        private static double CalculateCohesionForEveryVariable(CaDETClass parsedClass, IEnumerable<CaDETMember> memberMethods, IEnumerable<CaDETMember> memberProperties)
+        {
+            double cc = 0;
+
+            foreach (var field in parsedClass.Fields)
+            {
+                double methodsInvokingGivenField = memberMethods.Count(m => m.AccessedFields.Contains(field));
```

Comment:
The CaDETMember offers the following methods: https://github.com/Clean-CaDET/platform/blob/c3fa836fd7918b5b3c392525c4353c0d7a9482f7/RepositoryCompiler/CodeModel/CaDETModel/CodeItems/CaDETMember.cs#L50-L76

Would something like this (memberMethods.Sum(m => m.GetAccessedOwnFields().Count) + memberMethods.Sum(m => m.GetAccessedOwnAccessors().Count)) / memberMethods.Count give a good result?

If yes, we would also reduce the params of this function to 1 which is a plus.